btree/btree.c

```diff
@@ -3470,23 +3559,58 @@ static struct fkvv_head *fkvv_data(const struct nd *node)
 	return segaddr_addr(&node->n_addr);
 }
 
+static void *fkvv_dir_get(const struct nd *node)
+{
+	struct fkvv_head    *h   = fkvv_data(node);
+	return ((void *)h + sizeof(*h) + h->fkvv_dir_offset);
+}
+
+static void fkvv_dir_init(const struct nd *node)
+{
+	struct fkvv_head    *h = fkvv_data(node);
+	struct fkvv_dir_rec *dir;
+
+	h->fkvv_dir_offset  = (h->fkvv_nsize - sizeof(*h))/2;
+	h->fkvv_dir_entries   = 1;
+
+	dir = fkvv_dir_get(node);
+	dir[0].key_offset     = 0;
+	dir[0].key_size       = h->fkvv_dir_offset;
+	dir[0].val_offset     = h->fkvv_nsize - sizeof(*h) -
+				h->fkvv_dir_offset - sizeof(*dir);
+	dir[0].alloc_val_size = dir[0].val_offset;
+	dir[0].val_size       = 0;
+}
+
 static void fkvv_init(const struct segaddr *addr, int ksize, int vsize,
 		      int nsize, uint32_t ntype, uint64_t crc_type,
-		      uint64_t gen, struct m0_fid fid)
+		      uint64_t addr_type, uint64_t gen, struct m0_fid fid)
 {
-	struct fkvv_head *h       = segaddr_addr(addr);
+	struct fkvv_head    *h    = segaddr_addr(addr);
+	struct fkvv_dir_rec *dir;
 
 	M0_PRE(ksize != 0);
 	M0_SET0(h);
 
 	h->fkvv_seg.h_crc_type    = crc_type;
+	h->fkvv_seg.h_addr_type   = addr_type;
+	h->fkvv_dir_offset        = (nsize - sizeof(*h))/2;
+	h->fkvv_dir_entries         = 1;
 	h->fkvv_ksize             = ksize;
 	h->fkvv_nsize             = nsize;
 	h->fkvv_seg.h_node_type   = ntype;
 	h->fkvv_seg.h_gen         = gen;
 	h->fkvv_seg.h_fid         = fid;
 	h->fkvv_opaque            = NULL;
 
+	dir  = ((void *)h + sizeof(*h) + h->fkvv_dir_offset);
+	dir[0].key_offset     = 0;
+	dir[0].key_size       = h->fkvv_dir_offset;
+	dir[0].val_offset     = nsize - sizeof(*h) -
+				h->fkvv_dir_offset - sizeof(*dir);
+	dir[0].alloc_val_size = dir[0].val_offset;
+	dir[0].val_size       = 0;
+
```

Comment:
Can the above block of code be replaced using a function call to fkvv_dir_init()?